docs: missing service-linked role [DET-5253] (#2221)
determined-ai/determined: docs/topic-guides/deployment/aws-spot.txt

to not set ``spot_max_price`` and pay whatever the market price is. Your
mileage may vary, but at Determined, we have regularly seen 70% cost
reductions when using V100s, without specifying a ``spot_max_price``.

*****************
 Troubleshooting
*****************

Missing service-linked role
===========================

> Some users may encounter the following error in the Determined Master

Reviewer: ```suggestion
Some users may encounter the following error in the Determined master
```

Author: we totally need a linter for this

logs the first time they try to use spot instances:

.. code::

   AWS error while launching spot instances, AuthFailure.ServiceLinkedRoleCreationNotPermitted, The provided credentials do not have permission to create the service-linked role for EC2 Spot Instances.

> When this error occurs, please check `AWS documentation
> <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#service-linked-roles-spot-instance-requests>`__,
> or use AWS CLI to create the ``AWSServiceRoleForEC2Spot`` role:

Reviewer: If this is always a reasonable solution, maybe just go to it directly:
```suggestion
When this error occurs, please use the AWS CLI to create the ``AWSServiceRoleForEC2Spot`` role:
```
and move the doc link to the "For more details..." line.

Author: I thought so at first (which is why there's a leftover "for more details" after the code), but then decided to do this way. Upon looking at in it in the UI, it felt a bit too harsh to tell people to do `aws iam` outright. It's permissions stuff, it may be a risk, so they should really read the docs and have some awareness.

Reviewer: Ah, makes sense. The "or" sounds a little off and I kind of want them to be connected differently, though I'm not sure I can think of a good phrasing. Maybe:
```suggestion
When this error occurs, please check the `AWS documentation
<https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#service-linked-roles-spot-instance-requests>`__.
Most likely, you will need to use the AWS CLI to create the ``AWSServiceRoleForEC2Spot`` role:
```

.. code::

   aws iam create-service-linked-role --aws-service-name spot.amazonaws.com

> For more details, please see

Reviewer: Something's missing here.

Author: oops - that should've been removed.
